exchanges/request/request.go

```diff
@@ -150,7 +150,22 @@ func (r *Requester) doRequest(ctx context.Context, endpoint EndpointLimit, newRe
 			}
 		}
 
+		reporter := GlobalReporter
+		if reporter == nil {
+			// no global reporter defined, fallback to the request's
+			reporter = r.reporter
+			if reporter == nil {
+				// no custom reporter as well, fallback to the default
+				// that will look at the Verbose flag to decide wether
+				// to log or not
+				reporter = DefaultReporter(p.Verbose)
+			}
+		}
+
+		start := time.Now()
 		resp, err := r.HTTPClient.Do(req)
+		reporter.Latency(r.Name, p.Method, p.Path, time.Since(start))
```

Comment:
This reporting is optional behavior and I would prefer that things only are run if it is enabled

Reply:
`Latency` now will be done only when there's a defined reporter (either global or exchange scoped)